application/apps/indexer/addons/file-tools/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+// Idea is to get the file path from the user,
+// read the first chunks from the file and return String
+// 'binary' or 'text' as of now. Later we can add enum for file
+// type.
+
+use anyhow::Result;
+use std::{fs::File, io::Read, path::Path};
+
+const BYTES_TO_READ: usize = 10240;
+
+pub fn detect_file_type(file_path: &Path) -> String {
```

Comment:
Any IO related operations returns should return some `Result<>`. It's better then panic